<?xml version="1.0" ?>
<launch>
  <!--
      Launch file for the admittance controller
  -->

  <arg name="TOPIC_ARM_COMMAND" default="/joint_torque_controller/command_joint_torque"/>
  <arg name="TOPIC_WRENCH_STATE" default="/wrench"/>
  
  <param name="topic_cart_arm_state" value="/ee_state"/>
  <param name="topic_arm_state" value="/joint_torque_controller/joint_state"/>
  <param name="topic_arm_command" value="$(arg TOPIC_ARM_COMMAND)"/>
  <param name="topic_wrench_state" value="$(arg TOPIC_WRENCH_STATE)"/>
  
  <param name="topic_car_state" value="/base_pose_ground_truth"/>
    <param name="topic_car_imu" value="/imu_data"/>
  <param name="topic_car_command" value="/cmd_vel"/>

  <param name="topic_reference_cart_arm_state" value="/reference_ee_state"/>
  <param name="topic_reference_wrench_state" value="/reference_wrench"/>
  <param name="topic_reference_car_state" value="/reference_car_state"/>

  <param name="base_link" value="ur5ebase_link"/>
  <param name="end_link" value="ur5etool0"/>
  
  <rosparam file="$(find impedance)/config/ImpedanceParams.yaml" command="load"/>
  
  <node pkg="impedance" name="nokov_state_pub" type="nokov_state_pub.py"/>
  <node pkg="impedance" name="impedance" type="impedance_node" output="screen"/>
  <node pkg="impedance" name="reference_node" type="reference_node" output="screen"/>
</launch>
